apps/server/lib/bootstrap.ts

```diff
@@ -275,6 +275,147 @@ export const bootstrap = async (logContext: LogContext, options: any) => {
 		}),
 	);
 
+	await worker.replaceCard(
```

Comment:
If these types are working as expected, can you add them to the `jellyfish-worker` module so they get loaded by default?

Reply:
@iamsolankiamit Seems like we should be able to use the generic `thread` and `message` contracts for this and avoid adding new types entirely. I really don't like the idea of adding custom cruft specifically for Flowdock if we can use types that already exist. If you need something to flag these items as "being from Flowdock", you should be able to add something under `data` for the contracts, `data.mirrors=["https://flowdock.com/flow-foo/thread-bar"]` for example.

Reply:
@joshbwlng I tried using the thread and message contract, but they need an actor and we need extra fields like `flowdockThreadId`,  `flowdockFlow`. I figured we could keep flowdock archive separate from other JF data.

Reply:
If you are more inclined towards using the same contracts. do you have suggestions on how to incorporate the additional data? @joshbwlng

Reply:
@iamsolankiamit What do you mean exactly when you say "an actor", that can have a few meanings in Jellyfish. I'm guessing you mean the user contract used to create the thread/message contracts? And you should be able to store anything extra you have under `data`, `data.flowdock.threadId` for example. Is there a reason for splitting up the parts of the flowdock URL into things like `flow` and `threadId`? Could these be parts of a single field? If not, whats the blocker?

If we find that we really need to have Flowdock-specific contract types, they shouldn't be defined here or in the worker, we would bring the `plugin-flowdock` repo back to life and modify it to do most if not all of what you've done in this PR (could probably copy-paste most over).

Reply:
@joshbwlng didn't know we could add extra data like like that into a contract. I won't be able to mark the fields as required though.

about splitting the `flow` and `threadId`. Flow names have changed in past so urls might contain older names. threadId is unique though so it won't change.

Yes by actor I mean the user contract. Let me try and use the existing Threads and messages contracts

Reply:
@iamsolankiamit I'm not sure we need to be able to mark these flowdock-only fields as required, we won't be making any new ones or updating them post-import correct? We use `required` to ensure contract data shape validity on upserts, but thats less important in this case from what I can tell.

And I'll trust your judgement on the `flow` and `threadId` fields, I think I understand what you're talking about. My only concern was unnecessarily duplicating data.

Reply:
@joshbwlng I am unable to create messages without an actor (i.e. the user) since some of the users are deleted. is there a way to skip the required property? Should we create a special user called deleted-user and fallback to that user when we don't know who posted that message?

Reply:
@iamsolankiamit I see the issue. What actor are you using now for creating the flowdock archive threads/messages in this PR? We might want to stick with more or less what you have now, but bring the implementation out of this repo and into a new plugin (or the defunct flowdock plugin I guess).

Reply:
@joshbwlng In the flowdock archive contract I made. I didn't make the actor field as required, instead I'd pass the flowdock username directly and name in the contract itself. 

While creating the contracts jf would assign the api caller as the actor. I think we can skip this by using the admin context